common/src/main/java/com/microsoft/identity/common/internal/controllers/BaseController.java

```diff
@@ -375,6 +377,135 @@ protected void renewAccessToken(@NonNull final SilentTokenCommandParameters para
+                }
+
+            } else {
+                Logger.warn(TAG, "Invalid state, No token success or error response on the token result");
+            }
+        }
+    }
+
+    /*
+        Light version of:
+        protected renewAccessToken(@NonNull final SilentTokenCommandParameters parameters,
+                                   @NonNull final AcquireTokenResult acquireTokenSilentResult,
+                                   @SuppressWarnings(WarningType.rawtype_warning) @NonNull final OAuth2TokenCache tokenCache,
+                                   @SuppressWarnings(WarningType.rawtype_warning) @NonNull final OAuth2Strategy strategy,
+                                   @NonNull final ICacheRecord cacheRecord)
+
+        Diffs
+        1) acquireTokenSilentResult updates omitted.
+        2) All arguments derived from SilentTokenCommandParameters
+        3) New logic replacing old Access Token
+     */
+    public void renewAccessToken(@NonNull final SilentTokenCommandParameters parameters)
+            throws IOException, ClientException, ServiceException {
+        final String methodName = ":renewAccessToken";
+        Logger.info(
+                TAG + methodName,
+                "Renewing access token..."
+        );
+
+        //Extract strategy from parameters
+        final OAuth2StrategyParameters strategyParameters = new OAuth2StrategyParameters();
+        strategyParameters.setContext(parameters.getAndroidApplicationContext());
+        parameters.getAuthority().createOAuth2Strategy(strategyParameters);
+        final OAuth2Strategy strategy = parameters.getAuthority().createOAuth2Strategy(strategyParameters);
+
+        //Extract cache from parameters
+        final AccountRecord targetAccount = getCachedAccountRecord(parameters);
+        final AbstractAuthenticationScheme authScheme = parameters.getAuthenticationScheme();
+        final OAuth2TokenCache cache = parameters.getOAuth2TokenCache();
+
+        //Get cacheRecord from cache
+        final List<ICacheRecord> cacheRecords = cache.loadWithAggregatedAccountData(
+                parameters.getClientId(),
+                TextUtils.join(" ", parameters.getScopes()),
+                targetAccount,
+                authScheme
+        );
+        final ICacheRecord cacheRecord = cacheRecords.get(0);
+
+        Logger.info(
+                TAG + methodName,
+                "Attempting renewal of Access Token because it's refresh-expired. RefreshIn was expired at " + cacheRecord.getAccessToken().getRefreshOn() + ". Regular expiry is at " + cacheRecord.getAccessToken().getExpiresOn() + "."
+                        + "Currently executing acquireTokenSilent(..), SilentTokenCommand with CorrelationId: " + parameters.getCorrelationId()
+        );
+
+        //Get tokenResult
+        RefreshTokenRecord refreshTokenRecord = cacheRecord.getRefreshToken();
+        logParameters(TAG, parameters);
+        final TokenResult tokenResult = performSilentTokenRequest(
```

Comment:
You can probably refactor some of this code and then this method can just start at this line...

Reply:
Updated.